Enhance TrainPipelineSparseDist logging to help differentiate data loading patterns in train pipeline

cocoder39 opened the pull request:
Summary:
Observed inconsistent data loading behaviors in APS train_module_train_step. Expected 3 batches loaded on first invocation of train loop, but sometimes only 1 batch loading shows in trace ([link](https://www.internalfb.com/intern/sbdive/?id=tree%2Fttfb%2Fttfb_ai_lab_APS_mtml_ctr_cmf_rc1_baseline_gpu-f788555024-fbd033a0-89b2-4b72-b540-346901657b25-treatment-1&bucket=sbdive)) despite increasing trace frequency from 500ms to 50ms. Added logs to differentiate data loading patterns.


Perf impact: logs are added only when data loader is exhausted

Reviewed By: andywag

Differential Revision: D81418443

andywag reviewed (approved):
Review automatically exported from Phabricator review in Meta.

Commit:
Enhance TrainPipelineSparseDist logging to help differentiate data loading patterns in train pipeline (#3350)

Summary:
Pull Request resolved: https://github.com/pytorch/torchrec/pull/3350

Observed inconsistent data loading behaviors in APS train_module_train_step. Expected 3 batches loaded on first invocation of train loop, but sometimes only 1 batch loading shows in trace ([link](https://www.internalfb.com/intern/sbdive/?id=tree%2Fttfb%2Fttfb_ai_lab_APS_mtml_ctr_cmf_rc1_baseline_gpu-f788555024-fbd033a0-89b2-4b72-b540-346901657b25-treatment-1&bucket=sbdive)) despite increasing trace frequency from 500ms to 50ms. Added logs to differentiate data loading patterns.

Perf impact: logs are added only when data loader is exhausted

Reviewed By: andywag

Differential Revision: D81418443
